### DISCUSSION TITLE
How to load multi-modality data and multi-class mask easily?

### DISCUSSION THREAD

**Discussion by Bigsealion** [TorchIO-project/torchio#732] [Q&A] [answered]
Hi,
For example, I have 2 modality data with the same shape, and the mask is multi-class. I want to augment this data and input it into a PyTorch network. And I want to stack the 2 modalities as 2 channels, and set the mask as one-hot (means the output of the network will have the same channels with class).

Do I have to stack the data with dim-1 manually after augment? And do I have to apply one-hot to mask manually? Is there any default method in Torchio that can do this?

Thanks!

**Comment by romainVala** [comment, marked as answer]
Hi
did you see the different tutorial 
In this one, you get the answer with the `SubjectsDataset`
[general tutorial](https://colab.research.google.com/github/fepegar/torchio-notebooks/blob/main/notebooks/TorchIO_tutorial.ipynb)

and it is worth to have a look at the other ones found here
https://github.com/fepegar/torchio/tree/main/tutorials

**Reply by fepegar** [reply]
Additionally, please take a look at [`OneHot`](https://torchio.readthedocs.io/transforms/preprocessing.html#onehot).

**Reply by Bigsealion** [reply]
Thanks for your reply!
After reading some discussions and tutorials, I have found some solutions.

For loading multi-modality data (in same geomerty) to channels , using `img=tio.ScalarImage([img1_path, img2_path])` is easily.
And `OneHot `is also useful.
